warehouse/internal/loadfiles/loadfiles.go

```diff
@@ -192,155 +210,52 @@ func (lf *LoadFileGenerator) createFromStaging(ctx context.Context, job *model.U
-			if !ok {
-				return fmt.Errorf("receiving notifier channel closed")
-			}
-
-			lf.Logger.Infon("Received responses for staging files from notifier",
-				logger.NewIntField("startId", startId),
-				logger.NewIntField("endID", endId),
-				obskit.DestinationID(destID),
-				obskit.DestinationType(destType),
-			)
-			if responses.Err != nil {
-				return fmt.Errorf("receiving responses from notifier: %w", responses.Err)
-			}
-
-			var loadFiles []model.LoadFile
-			var successfulStagingFileIDs []int64
-			for _, resp := range responses.Jobs {
-				// Error handling during generating_load_files step:
-				// 1. any error returned by notifier is set on corresponding staging_file
-				// 2. any error effecting a batch/all the staging files like saving load file records to wh db
-				//    is returned as error to caller of the func to set error on all staging files and the whole generating_load_files step
-				var jobResponse WorkerJobResponse
-				if err := jsonrs.Unmarshal(resp.Payload, &jobResponse); err != nil {
-					return fmt.Errorf("unmarshalling response from notifier: %w", err)
-				}
-
-				if resp.Status == notifier.Aborted && resp.Error != nil {
-					lf.Logger.Errorf("[WH]: Error in generating load files: %v", resp.Error)
-					sampleError = errors.New(resp.Error.Error())
-					err = lf.StageRepo.SetErrorStatus(ctx, jobResponse.StagingFileID, sampleError)
-					if err != nil {
-						return fmt.Errorf("set staging file error status: %w", err)
-					}
-					continue
-				}
-				if len(jobResponse.Output) == 0 {
-					lf.Logger.Errorf("[WH]: No LoadFiles returned by worker")
-					continue
-				}
-				for _, output := range jobResponse.Output {
-					loadFiles = append(loadFiles, model.LoadFile{
-						TableName:             output.TableName,
-						Location:              output.Location,
-						TotalRows:             output.TotalRows,
-						ContentLength:         output.ContentLength,
-						StagingFileID:         output.StagingFileID,
-						DestinationRevisionID: output.DestinationRevisionID,
-						UseRudderStorage:      output.UseRudderStorage,
-						SourceID:              job.Upload.SourceID,
-						DestinationID:         job.Upload.DestinationID,
-						DestinationType:       job.Upload.DestinationType,
-						UploadID:              &job.Upload.ID,
-					})
-				}
-
-				successfulStagingFileIDs = append(successfulStagingFileIDs, jobResponse.StagingFileID)
-			}
-
-			if len(loadFiles) == 0 {
-				return nil
-			}
-
-			if err = lf.LoadRepo.Insert(ctx, loadFiles); err != nil {
-				return fmt.Errorf("inserting load files: %w", err)
-			}
-			if err = lf.StageRepo.SetStatuses(ctx, successfulStagingFileIDs, warehouseutils.StagingFileSucceededState); err != nil {
-				return fmt.Errorf("setting staging file status to succeeded: %w", err)
-			}
-			return nil
+			return lf.createUploadV2Jobs(gCtx, job, v2Files, publishBatchSize, uniqueLoadGenID)
 		})
 	}
 
-	if err := g.Wait(); err != nil {
+	if err = g.Wait(); err != nil {
```

Comment:
`g.Wait()` only returns first non-nil error. we don't want to loose the error if both routines return non-nil errors because one is v1 jobs and other is v2 jobs.

Reply:
`we don't want to loose the error`

Are you talking from a customer perspective or for debugging v2 errors? If latter, then I think the current implementation is fine since for most of the jobs, the type will be either only v1/v2 and not mixed.